src/main/java/com/hotels/beeju/core/BeejuCore.java

```diff
@@ -100,13 +103,44 @@ public BeejuCore(String databaseName, Map<String, String> preConfiguration, Map<
   }
 
   private void configureMisc() {
+    int webUIPort = getWebUIPort();
+
     // override default port as some of our test environments claim it is in use.
-    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_WEBUI_PORT, 0);
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_WEBUI_PORT, webUIPort);
     
     conf.setBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER, false);
     
     // Disable to get rid of clean up exception when stopping the Session.
     conf.setBoolVar(HiveConf.ConfVars.HIVE_SERVER2_LOGGING_OPERATION_ENABLED, false);
+
+    // Used to prevent "Not authorized to make the get_current_notificationEventId call" errors
+    setMetastoreAndSystemProperty(MetastoreConf.ConfVars.EVENT_DB_NOTIFICATION_API_AUTH, "false");
+
+    // Used to prevent "Error polling for notification events" error
+    conf.setTimeVar(HiveConf.ConfVars.HIVE_NOTFICATION_EVENT_POLL_INTERVAL, 0, TimeUnit.MILLISECONDS);
+
+    // Has to be added to exclude failures related to the HiveMaterializedViewsRegistry
+    conf.set(HiveConf.ConfVars.HIVE_SERVER2_MATERIALIZED_VIEWS_REGISTRY_IMPL.varname, "DUMMY");
+    System.setProperty(HiveConf.ConfVars.HIVE_SERVER2_MATERIALIZED_VIEWS_REGISTRY_IMPL.varname, "DUMMY");
+  }
+
+  private void setMetastoreAndSystemProperty(MetastoreConf.ConfVars key, String value) {
+    conf.set(key.getVarname(), value);
+    conf.set(key.getHiveName(), value);
+
+    System.setProperty(key.getVarname(), value);
+    System.setProperty(key.getHiveName(), value);
+  }
+
+  private int getWebUIPort() {
+    // Try to find a free port, if impossible return the default port 0 which disables the WebUI altogether
+    int DEFAULT_PORT = 0;
```

Comment:
```suggestion
    int defaultPort = 0;
```
(since it's no longer static and thus not a "constant")